db/db_impl_write.cc

```diff
@@ -62,6 +62,11 @@ Status DBImpl::WriteImpl(const WriteOptions& write_options,
                          WriteBatch* my_batch, WriteCallback* callback,
                          uint64_t* log_used, uint64_t log_ref,
                          bool disable_memtable) {
+  // The current implementation does not support sync with concurrenet writes
+  assert(!concurrent_writes_ || !write_options.sync);
+  if (concurrent_writes_ && write_options.sync) {
```

Comment:
`concurrent_wal_writes_` still feels confusing to me, as we don't actually make all WAL writes concurrent, but only prepare writes or prepare writes with one commit write. We aren't making commit WAL writes concurrent. Maybe it can be something like concurrent_prepare_wal_writes_, or concurrent_prepare_.